src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -818,16 +822,41 @@ def pivot(
         See detailed docstring in Snowpark DataFrame's pivot.
         """
         snowpark_dataframe = self.to_projected_snowpark_dataframe()
+
+        # Don't warn the user about our internal usage of private preview
```

Comment:
This seems wildly complex just to suppress a warning. injecting filters into logging just for this seems like overkill. I suggest one of the following:

1. Emit the warning only when a configuration variable is true, and we set it to false when snowpandas is initialized
2. Don't bother emitting the warning at all.

Reply:
+1 - since we're now integrated with the snowpark code, maybe we should add an extra kwarg that suppresses the warning to `prepare_pivot_arguments` and just set that in our code?

Reply:
I know this solution seems complex, but I thought about a few alternatives and thought that this solution made the most sense.

Here are the goals I think we should have:

1) We should warn Snowpark pandas users who are using Snowpark dynamic pivot, but not using it through Snowpark pandas, that the feature is experimental.
2) We should warn Snowpark users who are NOT snowpark pandas users that the Snowpark dynamic pivot feature is experimental.
3) We should **not** give the warning when users use the pivot via Snowpark pandas. Snowpark pandas users already get a warning that the entire Snowpark pandas feature is in private or public preview. Usually, they won't be aware that the pandas method they've called is using Snowpark pivot under the hood. 

Some solutions I have considered:

1) What I have currently. Fulfills all 3 goals.
2) As @sfc-gh-rdurrani  proposes, add a parameter to `prepare_pivot_arguments` that tells whether to give the warnings. The problem with this is that it only really makes sense to call `prepare_pivot_arguments` in the snowpark python `pivot()` implementation, and so then we have to pass this new parameter in to the user-exposed `pivot()` method or add a separate, private `pivot()` method with the extra parameter. I don't like either of those approaches-- we want to stick to using public interfaces of Snowpark dataframes, and I don't want to pollute the public interface of `pivot()` with this flag.
2) @sfc-gh-jkew 's solution 1). The problem is that if a Snowpark pandas user happens to also use dynamic pivot on a Snowpark dataframe, they will not get a warning. So this fulfills goals 2 and 3, but not goal 1.
3) @sfc-gh-jkew 's solution 2). This fulfills goal 3) but not goals 1) or 2).

about the complexity:

I know I implemented this change, but I don't think it's terribly complex. It's 30 lines contained in a method that does little else. There are 2 new classes but each is under 10 lines and I think they're each easy to understand.

If after this explanation this solution still seems unacceptable, I can go check with @sfc-gh-evandenberg to see if we can just get rid of the warnings completely.

Reply:
I don't think the first goal is particularly important in this case. We can just assume that if they are using snowpandas they are already using experimental features for PuPr.

Reply:
i pinged @sfc-gh-evandenberg to see if he's ok with skipping goal 1). waiting on response.

Reply:
Another way to keep track of flags in runtime context is Session.RuntimeConfig: https://github.com/snowflakedb/snowpark-python/blob/ed89373bc587e8f7e842a9393b39a0036bbb2ea3/src/snowflake/snowpark/session.py#L295. I also think warning filter is an overkill.

Reply:
Okay, I went with @sfc-gh-jkew 's option 1).